core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -0,0 +1,181 @@
+        } else {
+            actionImage.localImageName(config.dockerRegistry, config.dockerImagePrefix, Some(config.dockerImageTag))
+        }
+
+        DockerContainer.create(
+            tid,
+            image = image,
+            userProvidedImage = userProvidedImage,
+            memory = memory,
+            cpuShares = OldContainerPool.cpuShare(config),
+            environment = Map("__OW_API_HOST" -> config.wskApiHost),
+            network = config.invokerContainerNetwork,
+            name = Some(name))
+    }
+
+    val ack = (tid: TransactionId, activation: WhiskActivation) => {
+        implicit val transid = tid
+        producer.send("completed", CompletionMessage(tid, activation, s"invoker$instance")).andThen {
+            case Success(_) => logging.info(this, s"posted completion of activation ${activation.activationId}")
+        }
+    }
+
+    val store = (tid: TransactionId, activation: WhiskActivation) => {
+        implicit val transid = tid
+        logging.info(this, "recording the activation result to the data store")
+        WhiskActivation.put(activationStore, activation).andThen {
+            case Success(id) => logging.info(this, s"recorded activation")
+            case Failure(t)  => logging.error(this, s"failed to record activation")
+        }
+    }
+
+    val prewarmKind = "nodejs:6"
+    val prewarmExec = ExecManifest.runtimesManifest.resolveDefaultRuntime(prewarmKind).map { manifest =>
+        new CodeExecAsString(manifest, "", None)
+    }.get
+
+    val childFactory = (f: ActorRefFactory) => f.actorOf(ContainerProxy.props(containerFactory, ack, store))
+    val pool = actorSystem.actorOf(whisk.core.containerpool.ContainerPool.props(
+        childFactory,
+        OldContainerPool.getDefaultMaxActive(config),
+        activationFeed,
+        Some(PrewarmingConfig(2, prewarmExec, 256.MB))))
+
+    override def onMessage(msg: ActivationMessage)(implicit transid: TransactionId): Future[Unit] = {
+        require(msg != null, "message undefined")
+        require(msg.action.version.isDefined, "action version undefined")
+
+        val start = transid.started(this, LoggingMarkers.INVOKER_ACTIVATION)
+        val namespace = msg.action.path
+        val name = msg.action.name
+        val actionid = FullyQualifiedEntityName(namespace, name).toDocId.asDocInfo(msg.revision)
+        val tran = Transaction(msg)
+        val subject = msg.user.subject
+
+        logging.info(this, s"${actionid.id} $subject ${msg.activationId}")
+
+        // caching is enabled since actions have revision id and an updated
+        // action will not hit in the cache due to change in the revision id;
+        // if the doc revision is missing, then bypass cache
+        if (actionid.rev == DocRevision()) {
+            logging.error(this, s"revision was not provided for ${actionid.id}")
+        }
+        WhiskAction.get(entityStore, actionid.id, actionid.rev, fromCache = actionid.rev != DocRevision()).flatMap { action =>
```

Comment:
Hmm, we could. Not sure if it would be too beneficial though. I imagine the `Future` would then fail with different Exceptions, indicating a missing action or a failed deserialization? As we only need it in so little places, I'd leave it as is, at least for now.